lib/auth/auth.js

```diff
@@ -0,0 +1,56 @@
+'use strict'; // eslint-disable-line strict
+
+const errors = require('../errors');
+
+const AuthInfo = require('./AuthInfo');
+const authV2 = require('./v2/authV2');
+const authV4 = require('./v4/authV4');
+const constants = require('../constants');
+
+const auth = {};
+
+auth.setAuthHandler = handler => {
+    authV2.headerAuthCheck.setAuthHandler(handler);
+    authV2.queryAuthCheck.setAuthHandler(handler);
+    authV4.headerAuthCheck.setAuthHandler(handler);
+    authV4.queryAuthCheck.setAuthHandler(handler);
+    return auth;
+};
+
+auth.doAuth = (request, log, cb) => {
+    log.debug('running auth checks', { method: 'auth' });
+    const authHeader = request.headers.authorization;
+    // Check whether signature is in header
+    if (authHeader) {
+        log.trace('authorization header', { authHeader });
+        // TODO: Check for security token header to
+        // handle temporary security credentials
+        if (authHeader.startsWith('AWS ')) {
+            log.trace('authenticating request with auth v2 using headers');
+            authV2.headerAuthCheck.check(request, log, cb);
+        } else if (authHeader.startsWith('AWS4')) {
+            log.debug('authenticating request with Auth V4 using headers');
```

Comment:
As in previous comments, this is part of the original code from S3. I don't want to make this kind of changes in this PR with the aim of minimizing the diff. Let's leave it as it is for now ;)